src/ProjectTemplates/Shared/ProjectFactoryFixture.cs

```diff
@@ -35,33 +51,36 @@ public async Task<Project> GetOrCreateProject(string projectKey, ITestOutputHelp
         }
         return _projects.GetOrAdd(
             projectKey,
-            (key, outputHelper) =>
-            {
-                var project = new Project
-                {
-                    Output = outputHelper,
-                    DiagnosticsMessageSink = DiagnosticsMessageSink,
-                    ProjectGuid = Path.GetRandomFileName().Replace(".", string.Empty)
-                };
-                // Replace first character with a random letter if it's a digit to avoid random insertions of '_'
-                // into template namespace declarations (i.e. make it more stable for testing)
-                var projectNameSuffix = !char.IsLetter(project.ProjectGuid[0])
-                    ? string.Create(project.ProjectGuid.Length, project.ProjectGuid, (suffix, guid) =>
-                    {
-                        guid.AsSpan(1).CopyTo(suffix[1..]);
-                        suffix[0] = GetRandomLetter();
-                    })
-                    : project.ProjectGuid;
-                project.ProjectName = $"AspNet.{projectNameSuffix}";
-
-                var assemblyPath = GetType().Assembly;
-                var basePath = GetTemplateFolderBasePath(assemblyPath);
-                project.TemplateOutputDir = Path.Combine(basePath, project.ProjectName);
-                return project;
-            },
+            (_, outputHelper) => CreateProjectImpl(outputHelper),
             output);
     }
 
+    private Project CreateProjectImpl(ITestOutputHelper output)
+    {
+        var project = new Project
+        {
+            Output = output,
+            DiagnosticsMessageSink = DiagnosticsMessageSink,
+            ProjectGuid = Path.GetRandomFileName().Replace(".", string.Empty)
+        };
+        // Replace first character with a random letter if it's a digit to avoid random insertions of '_'
```

Comment:
Should we just write our own ProjectName name generator from a smaller set of chars, rather than relying on Guid.NewGuid, so we don't have to do this funky biz here?

Reply:
or alternatively just jam a random letter in front of the guid on creation time

Reply:
We could, although this is using `Path.GetRandomFileName()` rather than `Guid.NewGuid` here.

Reply:
I'll change this to append the letter to the front